### PR TITLE
Create a dedicated model for creating new players

### PR DESCRIPTION
**Pull request by fgheysels**
Changed the CreatePlayer operation to use a dedicated model instead of the `Player` entity model.

A new `NewPlayerDto` model has been introduced that defines the minimal required properties that are necessary to create a new Player.  (`PlayerDto` is not used since we do not want to specify a PlayerId).

When creating a new Player, we also check if there exists a Team for the specified TeamId.  If not, a BadRequest is returned.

### CODE REVIEW

**Comment by fgheysels** [comment]
I've seen in the code that we have a `ValidateModelAttribute` action-filter.  However, since all our controllers are decorated with the `ApiController` attribute, I believe that the `ValidateModelAttribute` filter is no longer required.

File: maturity-level-one/src/Controllers/PlayersController.cs [hunk 1]
```diff
@@ -74,10 +70,28 @@ public async Task<IActionResult> GetPlayer(int id)
...
-            await _worldCupRepository.CreatePlayer(player);
-            return CreatedAtRoute("get-player-byid", new { id = player.Id }, player);
+            var team = await _worldCupRepository.GetTeam(player.TeamId, includePlayers: false);
+
+            if (team == null)
+            {
+                return BadRequest($"The Team with Id {player.TeamId} does not exist.");
+            }
+
+            var playerEntity = new Player
```
Comment (samneirinck) [inline maturity-level-one/src/Controllers/PlayersController.cs]: Discussion point: where's the line between using Automapper and a mapping like this?

**Comment by samneirinck** [comment]
> 
> 
> I've seen in the code that we have a `ValidateModelAttribute` action-filter. However, since all our controllers are decorated with the `ApiController` attribute, I believe that the `ValidateModelAttribute` filter is no longer required.

Yes, would be good to remove it in a seperate PR.

File: maturity-level-one/src/Controllers/PlayersController.cs (hunk 1 above)
Comment (MassimoC) [reply maturity-level-one/src/Controllers/PlayersController.cs]: please open an issue to follow this discussion

File: maturity-level-one/src/Models/NewPlayerDto.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+namespace Codit.LevelOne.Models
+{
+    public class NewPlayerDto
+    {
+        [Required]
+        public string FirstName { get; set; }
+        [Required]
+        public string Description { get; set; }
+        [Required]
+        public bool IsTopPlayer { get; set; }
```
[the remark this answers is not in the document]
Comment (MassimoC) [reply maturity-level-one/src/Models/NewPlayerDto.cs]: please open an issue to follow this discussion

**Pull request merged**